pkg/modulewriter/modulewriter.go

```diff
@@ -348,20 +353,24 @@ func prepDepDir(depDir string, overwrite bool) error {
 		}
 	}
 
-	// clean up old dirs
+	if err := prepArtifactsDir(artifactsDir); err != nil {
+		return err
+	}
+
+	// remove any existing backups of deployment group
 	prevGroupDir := filepath.Join(ghpcDir, prevDeploymentGroupDirName)
```

Comment:
nit: I would put all the prevGroupDir in its own function (L360-L381) similar to what you have done with prepArtifactsDir. Something like prepPrevGroupDir.

Reply:
ack